### PR TITLE
feat(tabs): Add support for Indicators

### PR DESCRIPTION
**Pull request by damassi**
Addresses [MOPLAT-807]

This adds tab-based indicator support to our Tabs component. 

API: 

```tsx
<Tabs 
  indicators={[
    { tabName: "tab1", Component: () => <Text textAlign="right">hi</Text> }, 
    { tabName: "tab2", Component: () => <Text textAlign="right">hey</Text> }
  ]}
>
  <Tabs.Tab name="tab1" label="Tab 1">
    <Tabs.ScrollView>
      <Text>Tab 1 content</Text>
    </Tabs.ScrollView>
  </Tabs.Tab>
  <Tabs.Tab name="tab2" label="Tab 2">
    <Tabs.ScrollView>
      <Text>Tab 2 content</Text>
    </Tabs.ScrollView>
  </Tabs.Tab>
</Tabs>
```

https://github.com/artsy/palette-mobile/assets/236943/6c617495-8541-4c55-90b6-281dbb08c5c5

[MOPLAT-807]: https://artsyproduct.atlassian.net/browse/MOPLAT-807?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 
-ruby '>= 2.6.10'
+ruby '>= 2.6.5'
```
Comment (brainbicycle) [inline Gemfile:3]: *question, non-blocking*: why was this change required?

File: src/elements/Tabs/Tabs.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        </Tabs.Tab>
+        <Tabs.Tab name="Tab 2" label="Tab 2">
+          <Tabs.ScrollView>
+            <Text>Tab 2 content</Text>
+          </Tabs.ScrollView>
+        </Tabs.Tab>
+      </Tabs>
+    </Flex>
+  ))
+  .add("Tabs with Indicator", () => (
```
Comment (brainbicycle) [inline src/elements/Tabs/Tabs.stories.tsx:23]: nice :)

File: Gemfile (hunk 1 above)
Comment (damassi) [reply Gemfile:3]: so that its in sync with eigen / energy and we don't have to rbenv / switch ruby versions and so on

**Review by brainbicycle — APPROVED** [review]
nice one!

File: Gemfile (hunk 1 above)
Comment (brainbicycle) [reply Gemfile:3]: hmm we don't have ruby specified in Gemfile in eigen though right? https://github.com/artsy/eigen/blob/e7f4e53b645a6da721d759acba26598885e6f80d/Gemfile 
should we just remove and use .tool-versions instead?

**Comment by artsyit** [comment]
:rocket: PR was released in [`v11.0.35`](https://github.com/artsy/palette-mobile/releases/tag/v11.0.35) :rocket:
